numalogic/udfs/_metrics_utility.py

```diff
@@ -0,0 +1,97 @@
+from typing import Optional, Any
+
+from numaprom.monitoring.utility import get_metric
+from omegaconf import OmegaConf
+
+
+def create_metrics_from_config_file(config_file_path: str) -> dict[str, Any]:
+    config = OmegaConf.load(config_file_path)
+    metrics = {}
+    for metric_config in config.get("numalogic_metrics", []):
+        metric_type = metric_config["type"]
+        for metric in metric_config["metrics"]:
+            name = metric["name"]
+            description = metric.get("description", "")
+            label_pairs = metric.get("label_pairs", {})
+            static_label_pairs = metric.get("static_label_pairs", {})
+            metrics[name] = get_metric(
+                metric_type, name, description, label_pairs, static_label_pairs
+            )
+    return metrics
+
+
+class MetricsSingleton:
```

Comment:
we already have a singleton metaclass. Can't we use that?

Reply:
Little different. Saw that code but could not reuse